lib/cli-engine/config-array-factory.js

```diff
@@ -0,0 +1,894 @@
+        return names.reduce((map, name) => {
+            if (isFilePath(name)) {
+                throw new Error("Plugins array cannot includes file paths.");
+            }
+            const plugin = this._loadPlugin(name, importerPath, importerName);
+
+            map[plugin.id] = plugin;
+
+            return map;
+        }, {});
+    }
+
+    /**
+     * Load a given parser.
+     * @param {string} nameOrPath The package name or the path to a parser file.
+     * @param {string} importerPath The path to a config file that imports it.
+     * @param {string} importerName The name of a config file that imports it. This is just a debug info.
+     * @returns {DependentParser} The loaded parser.
+     */
+    _loadParser(nameOrPath, importerPath, importerName) {
+        debug("Loading parser %j from %s", nameOrPath, importerPath);
+
+        const { cwd } = internalSlotsMap.get(this);
+        const relativeTo = importerPath || path.join(cwd, ".eslintrc");
+
+        try {
+            const filePath = ModuleResolver.resolve(nameOrPath, relativeTo);
+
+            writeDebugLogForLoading(nameOrPath, relativeTo, filePath);
+
+            return new ConfigDependency({
+                definition: require(filePath),
+                filePath,
+                id: nameOrPath,
+                importerName,
+                importerPath
+            });
+        } catch (error) {
+
+            // If the parser name is "espree", load the espree of ESLint.
+            if (nameOrPath === "espree") {
+                debug("Fallback espree.");
+                return new ConfigDependency({
+                    definition: require("espree"),
+                    filePath: require.resolve("espree"),
+                    id: nameOrPath,
+                    importerName,
+                    importerPath
+                });
+            }
+
+            debug("Failed to load parser '%s' declared in '%s'.", nameOrPath, importerName);
+            error.message = `Failed to load parser '${nameOrPath}' declared in '${importerName}': ${error.message}`;
+
+            return new ConfigDependency({
+                error,
+                id: nameOrPath,
+                importerName,
+                importerPath
+            });
+        }
+    }
+
+    /**
+     * Load a given plugin.
+     * @param {string} nameOrPath The plugin name to load.
+     * @param {string} importerPath The path to a config file that imports it. This is just a debug info.
+     * @param {string} importerName The name of a config file that imports it. This is just a debug info.
+     * @returns {DependentPlugin} The loaded plugin.
+     * @private
+     */
+    _loadPlugin(nameOrPath, importerPath, importerName) {
+        debug("Loading plugin %j from %s", nameOrPath, importerPath);
+
+        const { additionalPluginPool, cwd } = internalSlotsMap.get(this);
+        let request, id;
+
+        if (isFilePath(nameOrPath)) {
+            request = id = nameOrPath;
```

Comment:
If I'm understanding correctly, would this allow the `plugins` array to contain a filepath? I think we should disallow this for now until we decide on something like `localPlugins`. (It could cause some ambiguity between slashes in a plugin name and slashes in a rule name.)

Reply:
Ah, yes, this `if` statement is unnecessary. Actually, the factory rejects file paths before it calls `this.loadPlugin()`.